contracts/src/bin/vendor.rs

```diff
@@ -65,38 +22,122 @@ fn main() {
 }
 
 fn run() -> Result<()> {
-    let artifacts = paths::contract_artifacts_dir();
-    fs::create_dir_all(&artifacts)?;
+    let vendor = Vendor::new()?;
 
-    log::info!("vendoring contract artifacts to '{}'", artifacts.display());
-    for (npm_path, local_path) in NPM_CONTRACTS {
-        log::info!("retrieving {}", npm_path);
-        let source = Source::npm(npm_path.to_string());
+    vendor
+        .full()
+        .github(
+            "BalancerV2Vault",
+            "balancer-labs/balancer-core-v2/v1.0.0-artifacts/\
+             deployments/artifacts/mainnet/Vault.json",
```

Comment:
Are we taking advantage of the fact that we use the mainnet abi for all networks.